src/Native/Unix/Common/pal_safecrt.h

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#pragma once
+
+#include <string.h>
+#include <assert.h>
+#include <errno.h>
+
+typedef int errno_t;
+
+static errno_t memcpy_s(void* dst, size_t sizeInBytes, const void* src, size_t count)
+{
+    assert(dst != nullptr);
+    assert(src != nullptr);
+    assert(sizeInBytes >= count);
+
+    const char* x = static_cast<const char*>(dst);
+    const char* y = static_cast<const char*>(src);
+    assert((x + count <= y) || (y + count <= x)); // should be using memmove if this fails
+
+    if (count > 0)
+    {
+        if (dst == nullptr)
+        {
+            return EINVAL;
+        }
+
+        if (src == nullptr || sizeInBytes < count)
+        {
+            memset(dst, 0, sizeInBytes);
+            return EINVAL;
```

Comment:
I can change it, but I was following what coreclr has, which is using EINVAL here:
https://github.com/dotnet/coreclr/blob/32f0f9721afb584b4a14d69135bea7ddc129f755/src/pal/src/safecrt/memcpy_s.cpp#L75